cmd/image-builder/github-workflow-integration.md

```diff
@@ -0,0 +1,156 @@
+   OIDC identity provider. This step ensures that the token is valid and has not been tampered with.
+
+5. **OCI Image build preparation**: The `oci-image-builder` pipeline uses the information from the OIDC token to clone the appropriate
+   source code for the building of the OCI image. It uses the information from the OIDC token and user defined parameters to
+   set the appropriate parameters for the image build and signing. The data from OIDC token is used to decide whether the OCI image should
+   be signed or not.
+
+6. **Building the OCI Image**: Once the source code is cloned and the build parameters are set, the `oci-image-builder` pipeline proceeds to
+   build the OCI image. The build process uses a kaniko executor as a build engine.
+
+7. **Pushing the OCI Image**: After the OCI image is built, it is then pushed to a specified OCI registry.
+
+8. **Signing the OCI Image**: If the build was triggered by a push event in GitHub, the `oci-image-builder` pipeline uses the `signify`
+   service to sign the OCI image.
+   This step ensures the integrity and authenticity of the OCI image.
+
+## GitHub OIDC Identity Token Claims
+
+The OIDC token issued by GitHub's OIDC identity provider contains several claims that are crucial for the `oci-image-builder` pipeline.
+These claims are used to identify the workflow triggering the build pipeline and to clone the appropriate version of the source code. This
+is essential for slc-29 compliance, as it ensures that the exact version of the code that was tested in the PR or for which the push was
+merged is built.
+
+The validity and integrity of OIDC token must be validated in `oci-image-builder` pipeline and fail pipeline execution if validation fails.
+Because OIDC token uses `JWT` format, it can be validated with a standard validation process against GitHub OIDC identity provider.
+
+### Workflow Identification Claims
+
+The OIDC token contains the following claims that can be used to identify the workflow that triggered the build pipeline. These include:
+
+- `iss`: The issuer of the token. This is always `https://token.actions.githubusercontent.com`.
+- `iat`: The time at which the token was issued.
+- `exp`: The time at which the token expires.
+- `jti`: A unique identifier for the token.
+- `nbf`: The time before which the token must not be accepted for processing.
+- `kid`: The key ID of the key used to sign the token.
+- `alg`: The algorithm used to sign the token.
+- `run_id`: The ID of the workflow run.
+- `run_number`: The number of the workflow run.
+- `actor`: The login of the user who initiated the workflow run.
+- `event_name`: The name of the event that triggered the workflow run.
+- `workflow`: The name of the workflow that triggered the workflow run.
+- `workflow_ref`: The git ref associated with the workflow file.
+- `repository`: The repository where the workflow run occurred.
+- `repository_owner`: The owner of the repository where the workflow run occurred.
```

Comment:
If you can provide values for these claims, please use bold font for them (like you do for [parameters and variables](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#code-font-and-bold-font)).

Reply:
Values for claims are set by GitHub OIDC identity provider.